docker-entrypoint.sh

```diff
@@ -1,63 +1,30 @@
 #!/usr/bin/env bash
 set -e
 
-source /funcs.sh
+USER="$1"
 
-if [ -d ".git" ]; then
-	SHA=$(git rev-parse HEAD)
-else
-	echo "Unable to determine SHA, failing."
-	exit 1
+if [ -z "$USER" ] ; then
+    echo "Invalid build user given, failing."
+    exit 1
 fi
 
-if [[ "z${IDENT_KEY}" == "z" ]]; then
-	echo "No deploy key set"
-else
-	mkdir -p ~/.ssh
-	echo "${IDENT_KEY}" > ~/.ssh/id_rsa
-	chmod 0600 ~/.ssh/id_rsa
-	FINGER_PRINT=$(ssh-keygen -E md5 -lf ~/.ssh/id_rsa | awk '{ print $2 }' | cut -c 5-)
-	echo "Using deploy key ${FINGER_PRINT}"
-fi
-
-# Docs mention the cache is at /tmp/cache. Make sure this is true!
-export COMPOSER_HOME="/tmp"
-
-# Provide simple fingerprint for detecting that a deployment is in progress
-export CLOUD_BUILD=1
-
-if [ "${PARSE_COMPOSER}" != "" ]; then
-	echo "Running parse_composer..."
-	parse_composer
-fi
-
-# Disable vendor-expose during composer install (CMS 4+)
-if [[ -f composer.lock && "$(cat composer.lock | jq '.packages[] | select(.name == "silverstripe/vendor-plugin")')" != "" ]]; then
-	disable_postinstall_vendor_expose
-fi
-
-composer_install
-
-if [ "${CLOUD_BUILD_DISABLED}" == "" ]; then
-	echo "Running cloud-build..."
+USER_ID=$(id -u "$USER")
+USER_GID=$(id -g "$USER")
 
-	# Run NPM/Yarn build script if the cloud-build command is defined
-	if [[ -f package.json && "`cat package.json | jq '.scripts["cloud-build"]?'`" != "null" ]]; then
-		set_node_version
+# Cache directory is mounted with root as owner so need to make it writeable by the non-root user. 
+chown -R $USER_ID:$USER_GID /tmp/cache
 
-		node_build
-	fi
+# SSH key is mounted in root ssh directory, but we need it in the non-root user's home directory. 
+cp ~/.ssh/id_rsa /home/"$USER"/.ssh/
```

Comment:
If there is no mounted key, this throws an error where previously it still progress. Can we add a check to ensure the `/root/.ssh/id_rsa` key exists before attempting to copy it.

Reply:
Fixed